Makefile

```diff
@@ -62,58 +101,77 @@ GRUB_MODULES = \
 	raid6rec \
 	video
 
-# Download the latest version of package $1 for architecture $(ARCH), unpacking
+all: boot install
+
+# Check if we're running under snapcraft. If not, we need to 'stage' some
+# packages by ourselves.
+ifdef SNAPCRAFT_PROJECT_NAME
+stage-package:
+	$(info Skipping staging of package $(package) under snapcraft)
+else
+# Download the latest version of package $package for architecture $(ARCH), unpacking
 # it into $(STAGEDIR). For example, the following invocation will download the
 # latest version of u-boot-rpi for armhf, and unpack it under STAGEDIR:
 #
-#  $(call stage_package,u-boot-rpi)
+#  $(MAKE) stage-package package=u-boot-rpi
 #
-define stage_package
+stage-package:
+	$(info Staging package $(package)...)
 	mkdir -p $(STAGEDIR)/tmp
-	( \
-		cd $(STAGEDIR)/tmp && \
-		apt-get download \
-			-o APT::Architecture=$(ARCH) $$( \
-				apt-cache \
-					-o APT::Architecture=$(ARCH) \
-					showpkg $(1) | \
-					sed -n -e 's/^Package: *//p' | \
-					sort -V | tail -1 \
-			); \
-	)
-	dpkg-deb --extract $$(ls $(STAGEDIR)/tmp/$(1)*.deb | tail -1) $(STAGEDIR)
-endef
-
-all: boot install
-
-boot:
-	# Check if we're running under snapcraft. If not, we need to 'stage'
-	# some packages by ourselves.
-ifndef SNAPCRAFT_PROJECT_NAME
-	$(call stage_package,grub-pc-bin)
-	$(call stage_package,grub-efi-amd64-signed)
-	$(call stage_package,shim-signed)
+	# setup chdist APT environment for SERIES-ARCH and run apt update
+	if [ ! -d  $(STAGEDIR)/tmp/chdist ]; then \
+	    chdist -d $(STAGEDIR)/tmp/chdist -a $(ARCH) create $(SERIES)-$(ARCH); \
+	    echo "deb $(ARCHIVE) $(SERIES) main" >$(STAGEDIR)/tmp/chdist/$(SERIES)-$(ARCH)/etc/apt/sources.list; \
+	    echo "deb $(ARCHIVE) $(SERIES)-updates main" >>$(STAGEDIR)/tmp/chdist/$(SERIES)-$(ARCH)/etc/apt/sources.list; \
```

Comment:
Should this include -security?

Reply:
Ah I did think about omitting -security because it's pocket-copied into -updates regularly, but I should at the very list have a comment explaining this; I've added it now in a new commit, do you think I should always list -security to be on the safe side?

Reply:
I would like to have -security always included. It's per convention.